packages/grafana-ui/src/components/Combobox/MultiCombobox.tsx

```diff
@@ -11,22 +11,52 @@ import { ValuePill } from './ValuePill';
 import { getMultiComboboxStyles } from './getMultiComboboxStyles';
 
 interface MultiComboboxBaseProps<T extends string | number> extends Omit<ComboboxBaseProps<T>, 'value' | 'onChange'> {
-  value?: string | Array<ComboboxOption<T>>;
-  onChange: (items?: Array<ComboboxOption<T>>) => void;
+  value?: string[] | Array<ComboboxOption<T>>;
+  onChange: (items?: T[]) => void;
 }
 
-type MultiComboboxProps<T extends string | number> = MultiComboboxBaseProps<T> & AutoSizeConditionals;
+export type MultiComboboxProps<T extends string | number> = MultiComboboxBaseProps<T> & AutoSizeConditionals;
 
 export const MultiCombobox = <T extends string | number>(props: MultiComboboxProps<T>) => {
-  const { options, placeholder } = props;
+  const { options, placeholder, onChange, value } = props;
+  const isAsync = typeof options === 'function';
 
-  const multiStyles = useStyles2(getMultiComboboxStyles);
+  const selectedItems = useMemo(() => {
+    if (!value || isAsync) {
+      //TODO handle async
+      return [];
+    }
 
-  const isAsync = typeof options === 'function';
+    if (!isComboboxOptions(value)) {
+      const resultingItems: Array<ComboboxOption<T> | undefined> = [];
+
+      for (const item of options) {
+        value.forEach((val, index) => {
+          if (val === item.value) {
+            resultingItems[index] = item; // Use index to keep order
+          }
+        });
+        if (resultingItems.length === value.length && !resultingItems.includes(undefined)) {
+          // We found all items for the values
+          break;
+        }
+      }
+      return resultingItems.filter((item) => item !== undefined); //TODO: Handle cases where value is not in options
+    }
+
+    return value;
+  }, [value, options, isAsync]);
+
+  const multiStyles = useStyles2(getMultiComboboxStyles);
 
   const [items, _baseSetItems] = useState(isAsync ? [] : options);
   const [isOpen, setIsOpen] = useState(false);
-  const [selectedItems, setSelectedItems] = useState<Array<ComboboxOption<T>>>([]);
+  //const [selectedItems, setSelectedItems] = useState<Array<ComboboxOption<T>>>(initialSelectedItems);
```

Comment:
Was this commented while adding the `selectedItems` on line 24? Do you still need this `selectedItems`?

Reply:
No, it was a little oopsie to leave the comment there 😅